### PR TITLE
[fix](schema change) bugfix for light schema change while with rollup

### LINKED ISSUE #15586: [Bug] [daily] be core when query after schema change
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master doris-0.0.0-trunk-a97f582

### What's Wrong?

jenkins daily failed case: https://ci-builds.apache.org/job/Doris/job/doris_daily_enable_vectorized/165/testReport/(root)/test_sys_rollup_scenario/test_schema_change_to_rollup_table/

core msg:
image

### What You Expected?

query ok

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by cambyzju**
# Proposed changes

Issue Number: close https://github.com/apache/doris/issues/15586

## Problem summary

Describe your changes.
this problem come from pr: https://github.com/apache/doris/pull/11494

After add column to rollup index, it also change column UniqueId inside base index. 

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
3. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
5. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
6. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by yiguolei** [comment]
Please add some regression tests.

File: fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java [hunk 1]
```diff
@@ -1106,18 +1106,20 @@ private void checkAndAddColumn(List<Column> modIndexSchema, Column newColumn, Co
             hasPos = true;
         }
 
-        newColumn.setUniqueId(newColumnUniqueId);
+        // newColumn may add to baseIndex or rollups, so we need copy before change UniqueId
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java:1109]: Does doris support add a column to rollup and the column is not in base table?

Comment (cambyzju) [reply fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java:1109]: not support.

now rollup columns are subset of base table.

**Comment by cambyzju** [comment]
> Please add some regression tests.

done

**Comment by Lchangliang** [comment]
LGTM

**Review by yangzhg — APPROVED** [review]
LGTM

**Pull request merged**